src/flashbar/collapsible-flashbar.tsx

```diff
@@ -112,7 +113,7 @@ export default function CollapsibleFlashbar({ items, ...restProps }: FlashbarPro
           const windowHeight = window.innerHeight;
           // Take the parent region into account if using the App Layout, because it might have additional margins.
           // Otherwise we use the Flashbar component for this calculation.
-          const outerElement = flashbar.parentElement?.parentElement || flashbar;
+          const outerElement = findUpUntil(flashbar, element => element.getAttribute('role') === 'region') || flashbar;
```

Comment:
Made this more robust as I realized it was not working well on Visual Refresh (it was grabbing one level further up to the correct one). This is because the DOM structure rendered by the app layout can be different on Visual Refresh than Classic —and actually also when notifications are stick vs when they aren't.